netsim/modules/srv6.py

```diff
@@ -1,29 +1,101 @@
 #
 # SRv6 transformation module
 #
+import typing
+
 from box import Box
+from ..augment import addressing,devices
 
 from . import _Module
 from ..utils import log
+from .. import data
 import ipaddress
 
-class SRV6(_Module):
+DEFAULT_VPN_AF: typing.Final[dict] = {
+  'ipv4': [ 'ibgp' ],
+  'ipv6': [ 'ibgp' ]
+}
 
-  def node_post_transform(self, node: Box, topology: Box) -> None:
+POOL_NAME = "srv6_locator"
 
-      # Could model this as another addressing pool too
-      locator = f'{topology.defaults.srv6.locator}:{node.id:x}::/64'
-      locator_net = ipaddress.IPv6Network(locator)
+"""
+Configures BGP address families for neighbors, including extended nexthop where needed
+"""
+def configure_bgp_for_srv6(node: Box, topology: Box) -> None:
+  srv6_af = node.get('srv6.af',{})
+  srv6_vpn = node.get('srv6.vpn')
+  for nb in node.get('bgp.neighbors',[]):
+    if 'ipv6' not in nb:           # Skip IPv4-only neighbors
+      continue
+    nb.ipv4_rfc8950 = True         # Enable extended next hops by default
```

Comment:
* I don't think your implementation supports inter-AS SRv6 SIDs, so maybe you should limit this to IBGP neighbors
* It does not make sense to enable extended next hops if we have an IPv4 address for the neighbor or if the activate flag will not be set.

Reply:
I added an EBGP test case, I may be missing some of the technical nuances but the pings seem to work. 
Made extended next hops dependent on having ipv4 AF without ipv4 transport